### PR TITLE
UIU-3043 - loader on externally uploaded profile picture(follow-up)

### PR DESCRIPTION
**Pull request by manvendra-s-rathore**
## Purpose
Fix loader for profile picture on UserDetail and UserEdit Screen, when profile picture is uploaded using external url in UserEdit Screen.
Fix is for below tickets :-
[UIU-3043](https://folio-org.atlassian.net/jira/software/c/projects/UIU/boards/76?selectedIssue=UIU-3043)
[UIU-3044](https://folio-org.atlassian.net/jira/software/c/projects/UIU/boards/76?selectedIssue=UIU-3044)

## Approach
1). Profile Picture can be uploaded in two ways :- local file upload and via external link.
    With external link upload, loader was not working as expected. This PR is to fix that issue.
2). A new common component(ProfilePicture) is created which can be used in EditUserProfilePicture and UserInfo components. 
 
## Screencast
https://github.com/folio-org/ui-users/assets/105021655/e6da29f2-0425-4b10-b51e-9934fc6fce6c

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/components/ProfilePicture/ProfilePicture.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      className={css.profilePlaceholder}
+      alt={intl.formatMessage({ id: 'ui-users.information.profilePicture' })}
+      src={imgSrc}
+      loader={<Loading />}
+    />
+  );
+};
+
+ProfilePicture.propTypes = {
+  profilePictureLink: PropTypes.string.isRequired,
```
Comment (Terala-Priyanka) [inline src/components/ProfilePicture/ProfilePicture.js:42]: I think "profilePictureLink" should not be a required props

File: src/components/ProfilePicture/ProfilePicture.test.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    isAValidURL.mockReturnValue(true);
+    render(
+      <ProfilePicture
+        profilePictureLink={profilePictureLink}
+      />
+    );
+    expect(Img).toHaveBeenCalled();
+    const renderedProfileImg = Img.mock.calls[0][0];
+    expect(renderedProfileImg.alt).toBe('ui-users.information.profilePicture');
+    expect(renderedProfileImg.src).toBe(profilePictureLink);
```
Comment (Terala-Priyanka) [inline src/components/ProfilePicture/ProfilePicture.test.js:51]: you could probably move the assertions to another unit test - "should render profile picture with correct props"?

It is good to have 1 test - 1 assertion.

File: src/components/UserDetailSections/UserInfo/UserInfo.test.js [hunk 3]
```diff
@@ -58,11 +62,13 @@ describe('Render userInfo component', () => {
     });
     it('should display profile picture', () => {
       renderUserInfo(props);
-      const profilePicture = screen.getByAltText('ui-users.information.profilePicture')
-      expect(profilePicture).toHaveAttribute('src', 'profilePictureLink')
+      expect(Img).toHaveBeenCalled();
+      const renderedProfileImg = Img.mock.calls[0][0];
+      screen.debug(expect(renderedProfileImg.alt).toBe('ui-users.information.profilePicture'));
```
Comment (Terala-Priyanka) [inline src/components/UserDetailSections/UserInfo/UserInfo.test.js:67]: you should probably consider removing this statement.

File: src/components/UserDetailSections/UserInfo/UserInfo.js [hunk 4]
```diff
@@ -146,14 +119,16 @@ const UserInfo = (props) => {
...
           hasViewProfilePicturePerm &&
             <Col xs={3}>
               <Row>
                 <Col xs={12}>
                   <KeyValue
                     label={<FormattedMessage id="ui-users.information.profilePicture" />}
-                    value={renderProfilePic()}
+                    value={<ProfilePicture
+                      profilePictureLink={profilePictureLink}
+                    />}
```
Comment (Terala-Priyanka) [inline src/components/UserDetailSections/UserInfo/UserInfo.js:131]: `value={<ProfilePicture profilePictureLink={profilePictureLink} />}`

**Review by Terala-Priyanka — CHANGES_REQUESTED** [review]

**Comment by manvendra-s-rathore** [comment]
Hey @folio-org/fe-tl-reviewers , can someone please review this PR.

**Review by Terala-Priyanka — APPROVED** [review]

**Review by UladzislauKutarkin — APPROVED** [review]

**Pull request merged**